_posts/2024-04-04-cheriot-supply-chain.markdown

```diff
@@ -0,0 +1,189 @@
+---
+layout: post
+title:  "CHERIoT and the supply chain"
+date:   2024-04-04T12:00+00:00
+categories: rtos supply-chain auditing
+author: "David Chisnall"
+---
+
+Late last week we learned that much of the world narrowly avoided a backdoor in SSH, introduced via the dependency on liblzma via libsystemd.
+A malicious actor introduced a backdoor into liblzma, which could exploit SSH via this dependency chain (introduced by Linux distributions, not in the upstream OpenSSH).
+This specific attack is not relevant to CHERIoT because it targets programs that have much bigger system requirements than the kinds of devices that we support, but the underlying concept is directly relevant.
+This is precisely the category of attack that CHERIoT was designed to protect against.
+
+## The origins of CHERI
+
+As a compression library, the interface of something like liblzma is quite simple.
+You provide it with buffers to compress, it provides buffers to decompress.
```

Comment:
Does it actually provide buffers or does the caller provide both in- and out-buffers?